src/main/java/net/minestom/server/network/packet/client/handshake/HandshakePacket.java

```diff
@@ -92,19 +90,18 @@ public void process(@NotNull PlayerConnection connection) {
                 return;
             }
         }
-
         switch (nextState) {
             case 1:
                 connection.setConnectionState(ConnectionState.STATUS);
                 break;
             case 2:
+                if (connection instanceof NettyPlayerConnection) {
+                    ((NettyPlayerConnection) connection).setProtocolVersion(protocolVersion);
```

Comment:
You can put the protocol in `refreshServerInformation`, there is no reason to have a new setter